### PR TITLE
Assignment2

### PR DESCRIPTION
**Pull request by AntonioGomez12**
website link: https://ashy-bush-0a1cb991e.3.azurestaticapps.net
up to date now! Sound works on localhost and chrome noticed I had to reset program to work sometimes, not sure why.

### CODE REVIEW

**Comment by AntonioGomez12** [comment]
updated more

File: wordle-web/src/components/LetterBase.vue [hunk 1]
```diff
@@ -1,5 +1,13 @@
 <template>
-  <v-btn :color="props.color">
+  <v-btn 
+     class="charButton"
```
Comment (mmoran314) [inline wordle-web/src/components/LetterBase.vue:3]: By default the buttons will have a shadow when they have some elevation, but I think I'd add this just to be safe
```suggestion
     class="charButton, shadow-5"
```

**Comment by tienv9** [comment]
Create a custom component to show available words ❌
Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ❌
Create unit tests for the validWords method ❌
Display the number of valid words based on the entered letters ❌
When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ❌
When a word is clicked, the word should be entered as the guess but not submitted ❌
Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ❌
**Missing**


Convert the sidebar to an App Bar ✅
Name of the app on the left with an icon ✅
Choose an icon for the app (from Material Design or somewhere on the web) ✅
Clicking the name and icon of the app take you to the home page ❌
**As of current it takes you to the wordle game page instead of home page**


Hamburger menu on the right ✅
Menu option for an About page ✅
Create an About page that has a few sentences about this class project and how great Meg is ✅
Make sure this has a back button ❌
**The back button returns to the Wordle page instead of the previous page.**


Add a settings dialog ❌
Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ❌
**This just goes to a setting page instead of opening a dialog**


Option for dark and light mode ✅
Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
The above two items must be implemented with built in Veutify features ✅
Style the main game page ✅
Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅
Add gradient to guesses blocks and keys (with CSS) ❌
Add drop shadow to keys (No CSS) ❌
**Missing gradient and drop shadow**


Make the app responsive (No CSS) ✅
Extra Credit
Add an appropriate sound when the UI letter buttons are clicked (5 points)❌
Add creative styling following good UI/UX practices✅


Overall, everything that works looks okay other than the NavBar is on top of the gameboard which needs to be fixed, and the Settings should be a dialog or at least something that you can open anywhere on your website instead of going to a setting page.

**Comment by kinpark** [comment]
1. Create a custom component to show available words ❌
- Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ❌
- Create unit tests for the validWords method ❌
- Display the number of valid words based on the entered letters ❌
- When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ❌
- When a word is clicked, the word should be entered as the guess but not submitted ❌
- Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ❌

2. Convert the sidebar to an App Bar ✅
- Name of the app on the left with an icon ✅
- Choose an icon for the app (from Material Design or somewhere on the web) ✅
- Clicking the name and icon of the app take you to the home page ✅
- Hamburger menu on the right ✅
- Menu option for an About page ✅
  - Create an About page that has a few sentences about this class project and how great Meg is ✅
  - Make sure this has a back button ✅

3. Add a settings dialog ❌✅
- Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ✅
- Option for dark and light mode ✅
- Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ❌
- The above two items must be implemented with built in Veutify features ✅

4. Style the main game page ❌✅
- Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅
- Add gradient to guesses blocks and keys (with CSS) ❌
- Add drop shadow to keys (No CSS) ❌
- Make the app responsive (No CSS) ✅

5. Include a link to your Azure App in your PR comments.

## Extra Credit

- Add an appropriate sound when the UI letter buttons are clicked (5 points)
- Add creative styling following good UI/UX practices
  - Look at products by Google or some of the vuetify components for good UX/UI ideas
  - https://m2.material.io/design/guidelines-overview

File: wordle-web/src/App.vue [hunk 2]
```diff
@@ -1,28 +1,15 @@
...
-        <RouterLink to="/">Home</RouterLink> | <RouterLink to="/wordle">Wordle</RouterLink> |
-        <RouterLink to="/about">About</RouterLink>
-      </nav>
-      <v-btn @click="setInverseTheme"> Inverse Theme </v-btn>
-      <v-btn @click="setDarkTheme"> Dark Theme </v-btn>
-    </div>
+    
   </header>
-
+  <nav-bar></nav-bar>
```
Comment (mmoran314) [inline wordle-web/src/App.vue:5]: This nav bar ends up covering some of the game boxes. Would add something to make sure the bar doesn't cover any important content.

**Comment by kinpark** [comment]
You haven't implemented the validWords component yet. The App Bar and the Settings Dialog looks great, though the color schemes seem to only be implemented for the dark theme, and should also be available in light mode. The main game page is looking good, though you still need to implement the gradient and drop shadow to the keys and guess blocks, and the nav bar obscures the top items in the body of whichever page you navigate to on your site.

File: wordle-web/src/views/SettingsView.vue
Comment (mmoran314) [inline wordle-web/src/views/SettingsView.vue:1]: I believe this should be a dialog box that can appear on any page rather than a page itself.

**Review by mmoran314 — COMMENTED** [review]
Create a custom component to show available words ❌
Create/extend a word list component with a method called validWords that returns an array of valid words based on current guesses ❌
Create unit tests for the validWords method ❌
Display the number of valid words based on the entered letters ❌
When the number is clicked, show a dialog with all the available words with scrolling if there are too many to show on the screen. ❌
When a word is clicked, the word should be entered as the guess but not submitted ❌
Add a hover over for the word so it is obvious that it can be clicked. (No CSS) ❌
Convert the sidebar to an App Bar ✅
Name of the app on the left with an icon ✅
Choose an icon for the app (from Material Design or somewhere on the web) ✅
Clicking the name and icon of the app take you to the home page ❌ 
It takes you to the game page instead, just needs a quick fix to change where it leads

Hamburger menu on the right ✅
Menu option for an About page ✅
Create an About page that has a few sentences about this class project and how great Meg is ✅
Make sure this has a back button ✅
Add a settings dialog ❌✅ 
There is a settings page with a dialog box, but I believe a dialog should be a box that opens within the page you are currently on

Add a settings selection to the App Bar's menu with a gear icon. This should open this dialog regardless of where you are on the site ❌✅ 
Again, this takes you to the settings page rather than opening a dialog on any given page.

Option for dark and light mode ✅
Develop two additional color schemes (with creative names) and allow the user to change to them. Schemes should look good in both light and dark mode ✅
The above two items must be implemented with built in Veutify features ✅
Style the main game page ✅
Make it look like [wordle.com](https://www.nytimes.com/games/wordle/index.html) or the Wordle mobile app ✅

There are a few things that can make it look more like the Wordle page such as adding a backspace key to the keyboard

Add gradient to guesses blocks and keys (with CSS) ❌
Add drop shadow to keys (No CSS) ✅
Make the app responsive (No CSS) ✅
Include a link to your Azure App in your PR comments.

Still a few things left to do but what is done looks really nice.

**Comment by AntonioGomez12** [comment]
updated site, should meet all specs now

**Comment by AntonioGomez12** [comment]
Buttons make sound in chrome also in localhost, sometimes it stops working not sure why

**Review by mmwoodfo — COMMENTED** [review]
image
Your keybaord colors dont' seem to update correctly. When a letter is correctly guessed, it should change the keyboard color to the 'correct' color.

File: wordle-web/src/scripts/wordsService.ts [hunk 3]
```diff
@@ -7,9 +11,66 @@ export abstract class WordsService {
     return this.#words.includes(word)
   }
 
-  static validWords(): Array<string> {
+  static validWords(guess: Word, secretWord: string) {
```
Comment (mmwoodfo) [inline wordle-web/src/scripts/wordsService.ts:14]: This function doesn't seem to always update the word list

**Pull request closed without merging**